cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConfiguration.java

```diff
@@ -50,4 +66,119 @@ private KafkaConfiguration(String configuration, List<String> forbiddenOptions)
     public static KafkaConfiguration unvalidated(String string) {
         return new KafkaConfiguration(string, emptyList());
     }
+
+    /**
+     * Validate the configs in this KafkaConfiguration returning a list of errors.
+     * @param kafkaVersion The broker version.
+     * @return A list of error messages.
+     */
+    public List<String> validate(KafkaVersion kafkaVersion) {
+        List<String> errors = new ArrayList<>();
+        Map<String, ConfigModel> models = readConfigModel(kafkaVersion);
+        for (Map.Entry<String, String> entry: asOrderedProperties().asMap().entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            // TODO, special parsing for "listener.security.protocol.map"
```

Comment:
Since you do the special parsing for `listener.security.protocol.map` ... how does this deal with the forbidden options? Because that is one of the forbidden options.

Reply:
Currently the validation doesn't know about the forbidden options. TBH I put this code in for dealing with the `listener.name.([^.]*).(ssl|sasl).*` without thinking enough about the forbidden options. Currently we're doing this validation on the options as given by the user (after forbidden option filtering), so there's no benefit from having this code for validating listener config specifically. I guess I'm struggling to see that it offers any benefit (either right now or in the foreseeable future), so I guess I should remove it.

Reply:
Could be useful if we use it ever for configuration of the listeners in our code. But until then it should be not needed I guess.

Reply:
It's not completely clear whether it is better to validate the whole configuration (including what's generated by the operator itself). The only difference I can see it making at the moment would be catching bugs in the operator. It might start being more useful if we allowed users the ability to configure per-listener configs directly in `Kafka.spec.kafka.configs`. It should be harmless to keep this code for now if you think we're likely to do that?

Reply:
I think there is no harm in validating what the oerator it self generates. So IMHO it might be useful in the future.

Reply:
I was a bit rusty on what the operator does. The CO doesn't actually add anything to the configuration which is passed to the pod. All the bits defined or implied from elsewhere in the CR are passed in other env vars and the script in the pod assembles everything (since some of those configs depend on variables which are only available to the pod). While some of the configs defined in the scripts could be defined by the operator (e.g. some are constants), there's little benefit in validating just a few more things which we already know are valid.

Reply:
Well, I think in the KAfkaConfiguation the listeners would never be there. But in other parts of code we might eventually configure things differently (through the APIs and not env.vars which force restart.)